lua/modules/linear_bolzmann_solvers/discrete_ordinates_solver/lbs_compute_leakage.cc

```diff
@@ -49,4 +48,66 @@ chiLBSComputeLeakage(lua_State* L)
   return 1;
 }
 
+RegisterLuaFunction(ComputeLeakage, lbs, ComputeLeakage);
+
+int
+ComputeLeakage(lua_State* L)
+{
+  const auto fname = "lbs.ComputeLeakage";
+  const auto num_args = lua_gettop(L);
+
+  // Get the solver
+  LuaCheckNilValue(fname, L, 1);
+  const auto solver_handle = lua_tointeger(L, 1);
+  const auto& solver = opensn::Chi::GetStackItem<opensn::lbs::DiscreteOrdinatesSolver>(
+    opensn::Chi::object_stack, solver_handle, fname);
+
+  // Get the boundaries to parse
+  std::vector<uint64_t> bndry_ids;
+  if (num_args > 1)
+  {
+    LuaCheckTableValue(fname, L, 2);
+
+    // Define mapping from boundary name to id
+    const std::map<std::string, uint64_t> bndry_map = {
+      {"xmax", 0}, {"xmin", 1}, {"ymax", 2}, {"ymin", 3}, {"zmax", 4}, {"zmin", 5}};
```

Comment:
This map is already in [`lbs_solver.cc`](https://github.com/Open-Sn/opensn/blob/9f4c021ea8564f650895129919664834434e2f6b/modules/linear_boltzmann_solvers/a_lbs_solver/lbs_solver.cc#L655-L656), but as a local variable. If the client code needs to be able to use it, then the local variable `supported_bndry_names` from `SetBoundaryOptions` should be a public static of the `LBSSolver` and renamed to something like `boundary_id_map` (or `supported_boundary_id_name`).

Reply:
I agree with this. I think in an ideal world, there is a public static member `std::vector<std::string> supported_boundary_names` and the mappings are internal. The natural inclination would be to make the `ComputeLeakage` routine accept a vector of strings. This doesn't jive with future plans @ragusa and I have discussed to have arbitrary boundary IDs to support more complex boundary conditions (e.g. prescribing boundary conditions to portions of boundaries). Opened issue #70 for this. I will make the maps public for the time being.